docs/architecture/index.rst

```diff
@@ -1,25 +1,586 @@
+======================
+
+Determined comprises a single *master* and one or more *agents* in a cluster environment. The master
+is a single, non-GPU node that manages the cluster.
+
+The master keeps experiment metadata in a PostgreSQL database, which can be queried using the WebUI
+or CLI. You can interact with the master, directly, but cannot interact with agents.
+
+Master Node Scope of Responsibility
+-----------------------------------
+
+Master node responsibilities:
+
+-  Stores experiment, trial, and workload metadata.
+-  Schedules and dispatch experiments to agents.
+-  Dynamically provisions and deprovisions on-premises and cloud agents.
+-  Advances experiment, trial, and workload state machines.
+-  Responds to CLI commands.
+-  Hosts the WebUI, which is primarily used to monitor experiments.
+-  Serves the REST API.
+
+The master is also responsible for the following training-specific searcher, metrics, checkpointing,
+and scheduling functionality:
+
++---------------+----------------------------------------------------------------------+
+| Function      | Description                                                          |
++===============+======================================================================+
+| Searcher      | The *Searcher* implements a hyperparameter search algorithm and is   |
+|               | responsible for coordinating the work of all experiment trials.      |
++---------------+----------------------------------------------------------------------+
+| Metrics       | *Metrics* provides persistent storage of metrics reported by Trials. |
++---------------+----------------------------------------------------------------------+
+| Checkpointing | *Checkpointing* captures the training state at a given time in the   |
+|               | *model registry*.                                                    |
++---------------+----------------------------------------------------------------------+
+| Scheduling    | *Scheduling* schedules jobs to run, ensuring that all of the compute |
+|               | resources required for a job are available before the job launches.  |
++---------------+----------------------------------------------------------------------+
+
+Agent Node Scope of Responsibility
+----------------------------------
+
+Agent node responsibilities:
+
+-  Discovers local computing devices and sends device/slot metadata to the master.
+-  Runs workloads at the request of the master.
+-  Monitors containers and sends container information to the master.
+-  For a *trial runner*, which runs a trial in a containerized environment, reports trial runner
+   states to the master.
+
+The agent manages a number of GPU or CPU devices, which are referred to as *slots*. There is
+typically one agent per compute server and the active experiment volume dictates the number of
+agents needed.
+
+Agents communicate only with the master and state information is not maintained on the agent.
```

Comment:
Maybe:

Agents communicate only with the master and ~state information is not maintained on the agent~ do not persist any information across restarts.

What I mean is, the agent process is very stateful... just when a new agent process starts up, it doesn't start with any state that was left for it by the last agent process.